File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/MaskedTextBox/Features/AllowAndResetPromptPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.MaskedTextBox.Features
{
    public partial class AllowAndResetPromptPage : UserControl
    {
        public AllowAndResetPromptPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the CheckedChanged event of the mobjResetCheckBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjResetCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            mobjMaskedTextBox.ResetOnPrompt = mobjResetCheckBox.Checked;
        }

        /// <summary>
        /// Handles the CheckedChanged event of the mobjAllowCheckBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjAllowCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            mobjMaskedTextBox.AllowPromptAsInput = mobjAllowCheckBox.Checked;
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/MaskedTextBox/Features/AllowAndResetPromptPage.vb
Namespace CompanionKit.Controls.MaskedTextBox.Features

	Public Class AllowAndResetPromptPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

        End Sub

        ''' <summary>
        ''' Handles the CheckedChanged event of the mobjResetCheckBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjResetCheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles mobjResetCheckBox.CheckedChanged
                mobjMaskedTextBox.ResetOnPrompt = mobjResetCheckBox.Checked
        End Sub

        ''' <summary>
        ''' Handles the CheckedChanged event of the mobjAllowCheckBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjAllowCheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles mobjAllowCheckBox.CheckedChanged
                mobjMaskedTextBox.AllowPromptAsInput = mobjAllowCheckBox.Checked
        End Sub

	End Class

End Namespace